solidago/src/solidago/pipeline/pipeline.py

```diff
@@ -185,20 +185,20 @@ def __call__(
         logger.info(f"Pipeline 1. Terminated in {np.round(start_step2 - start_step1, 2)} seconds")
         if output is not None:
             output.save_trust_scores(trusts=users)
+        
+        logger.info(f"Pipeline 2. Learning preferences with {str(self.preference_learning)}")
+        user_models = self.preference_learning(judgments, users, entities, init_user_models)
+        start_step3 = timeit.default_timer()
+        logger.info(f"Pipeline 2. Terminated in {np.round(start_step3 - start_step2, 2)} seconds")
+        raw_scorings = user_models
             
-        logger.info(f"Pipeline 2. Computing voting rights with {str(self.voting_rights)}")
+        logger.info(f"Pipeline 3. Computing voting rights with {str(self.voting_rights)}")
         # WARNING: `privacy` may contain (user, entity) even if user has expressed no judgement
         # about the entity. These users should not be given a voting right on the entity.
         # For now, irrelevant privacy values are excluded in `input.get_pipeline_kwargs()`
-        voting_rights, entities = self.voting_rights(users, entities, vouches, privacy)
-        start_step3 = timeit.default_timer()
-        logger.info(f"Pipeline 2. Terminated in {np.round(start_step3 - start_step2, 2)} seconds")
-        
-        logger.info(f"Pipeline 3. Learning preferences with {str(self.preference_learning)}")
-        user_models = self.preference_learning(judgments, users, entities, init_user_models)
+        voting_rights, entities = self.voting_rights(users, entities, vouches, privacy, user_models)
```

Comment:
This PR could be an opportunity to solve this warning for good: maybe we could specify that any `VotingRightsAssignment` implementation should ensure that no voting right is assigned to a (user, entity) that has no score in `user_models`?